pkg/provider/onboardbase/client.go

```diff
@@ -0,0 +1,224 @@
+	}
+	objectKey := types.NamespacedName{
+		Name:      credentialsSecretName,
+		Namespace: c.namespace,
+	}
+	// only ClusterStore is allowed to set namespace (and then it's required)
+	if c.storeKind == esv1beta1.ClusterSecretStoreKind {
+		if c.store.Auth.OnboardbaseAPIKey.Namespace == nil {
+			return fmt.Errorf(errInvalidClusterStoreMissingOnboardbaseAPIKeyNamespace)
+		}
+		objectKey.Namespace = *c.store.Auth.OnboardbaseAPIKey.Namespace
+	}
+
+	err := c.kube.Get(ctx, objectKey, credentialsSecret)
+	if err != nil {
+		return fmt.Errorf(errFetchOnboardbaseAPIKeySecret, err)
+	}
+
+	onboardbaseAPIKey := credentialsSecret.Data[c.store.Auth.OnboardbaseAPIKey.Key]
+	if (onboardbaseAPIKey == nil) || (len(onboardbaseAPIKey) == 0) {
+		return fmt.Errorf(errMissingOnboardbaseAPIKey, c.store.Auth.OnboardbaseAPIKey.Key, credentialsSecretName)
+	}
+	c.onboardbaseAPIKey = string(onboardbaseAPIKey)
+
+	onboardbasePasscode := credentialsSecret.Data[c.store.Auth.OnboardbasePasscode.Key]
+	if (onboardbasePasscode == nil) || (len(onboardbasePasscode) == 0) {
+		return fmt.Errorf(errMissingOnboardbasePasscode, c.store.Auth.OnboardbasePasscode.Key, credentialsSecretName)
+	}
+
+	c.onboardbasePasscode = string(onboardbasePasscode)
+
+	return nil
+}
+
+func (c *Client) Validate() (esv1beta1.ValidationResult, error) {
+	timeout := 15 * time.Second
+	clientURL := c.onboardbase.BaseURL().String()
+
+	if err := utils.NetworkValidate(clientURL, timeout); err != nil {
+		return esv1beta1.ValidationResultError, err
+	}
+
+	if err := c.onboardbase.Authenticate(); err != nil {
+		return esv1beta1.ValidationResultError, err
+	}
+
+	return esv1beta1.ValidationResultReady, nil
+}
+
+func (c *Client) DeleteSecret(ctx context.Context, remoteRef esv1beta1.PushRemoteRef) error {
+	return fmt.Errorf("not implemented")
+}
+
+func (c *Client) PushSecret(ctx context.Context, value []byte, remoteRef esv1beta1.PushRemoteRef) error {
+	return fmt.Errorf("not implemented")
+}
+
+func (c *Client) GetSecret(_ context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	request := dClient.SecretRequest{
+		Project:     c.project,
+		Environment: c.environment,
+		Name:        ref.Key,
+	}
+
+	secret, err := c.onboardbase.GetSecret(request)
+	if err != nil {
+		return nil, fmt.Errorf(errGetSecret, ref.Key, err)
+	}
+
+	return []byte(secret.Value), nil
+}
+
+func (c *Client) GetSecretMap(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) (map[string][]byte, error) {
+	data, err := c.GetSecret(ctx, ref)
+	if err != nil {
+		return nil, err
+	}
+
+	kv := make(map[string]json.RawMessage)
+	err = json.Unmarshal(data, &kv)
+	if err != nil {
+		return nil, fmt.Errorf(errUnmarshalSecretMap, ref.Key, err)
+	}
+
+	secretData := make(map[string][]byte)
+	for k, v := range kv {
+		var strVal string
+		err = json.Unmarshal(v, &strVal)
+		if err == nil {
+			secretData[k] = []byte(strVal)
+		} else {
+			secretData[k] = v
+		}
+	}
```

Comment:
with `property` we want to enable users to fetch multiple nested secrets. if the following secret is stored in the provider:
```
{ "foo" : {"nested" : { "bar" : "yes" } } }
```
an externalsecret below should produce a secret with `bar: yes`:

```
apiVersion: external-secrets.io/v1beta1
kind: ExternalSecret
metadata:
  name: service-name-secrets
spec:
  refreshInterval: 10m
  secretStoreRef:
    name: onboardbase-external-secret-store
    kind: SecretStore
  target:
    name: service-name-secrets
  dataFrom:
  - extract:
      key: my-nested-secret
      property: "foo.nested.bar"
```

It should still be compatible with secrets in the form of `file.json` in the key (you can see hashicorp vault provider implementation using `gjson` as a reference.

Reply:
We don't support nested secrets at the moment.

We have this kind of data store at the moment.
```
{
  key[string]: value[string]
}
```

Reply:
what I mean is that an user can load a JSON as the secret value. something like

```
{
  key:  "{\"foo\": \"bar\"}
 }
 ```

if even so this is still not supported, you should properly error it out as a not-supported feature.

Reply:
This should now be supported. 

https://github.com/external-secrets/external-secrets/pull/2128/files#diff-689f029bc38fc7365e7cb9b7003f062ae949dfef35e23a86c72631da627e88a8R146-R155